core/trino-parser/src/main/java/io/trino/sql/tree/SetColumnType.java

```diff
@@ -30,17 +30,7 @@
     private final DataType type;
     private final boolean tableExists;
 
-    public SetColumnType(QualifiedName tableName, Identifier columnName, DataType type, boolean tableExists)
-    {
-        this(Optional.empty(), tableName, columnName, type, tableExists);
-    }
-
-    public SetColumnType(NodeLocation location, QualifiedName tableName, Identifier columnName, DataType type, boolean tableExists)
-    {
-        this(Optional.of(location), tableName, columnName, type, tableExists);
-    }
-
-    private SetColumnType(Optional<NodeLocation> location, QualifiedName tableName, Identifier columnName, DataType type, boolean tableExists)
+    public SetColumnType(Optional<NodeLocation> location, QualifiedName tableName, Identifier columnName, DataType type, boolean tableExists)
```

Comment:
location is always known (except for `io.trino.execution.TestSetColumnTypeTask#executeSetColumnType` where you can fill it with dummy value), so no need for optional?